### ISSUE TITLE
Deleted attachments are not deleted directly in .attachments

### ISSUE THREAD

**Issue by devnull4242** [nextcloud/text#6503]
I don't know whether the current behaviour is intentional or not. But I could imagine that it could be somewhat problematic in terms of GDPR?

I can attach files in Nextcloud Text. These are then saved in the .attachments.805550 folder, for example. I would have expected the attachments to be deleted directly from .attachments.805550 (or move to trash bin) when they are deleted from the Markdown document. This is not the case. The .attachments.805550 folder and files are only deleted when the Markdown, e.g. document.md, is deleted.

Is that what you want? Is this possibly a problem with GDPR? Shouldn't files be deleted directly? I realise that there are of course dependencies on the recycle bin in Nextcloud.

**Comment by mejo-** [comment]
Dear @devnull4242, thanks for your report.

Text keeps deleted attachments in the `.attachments` folder for the life of the current editing session, to be able to undo the removal later on.

Once all session members left the session and it gets cleaned up by the cronjob, dangling attachments that are no longer referenced in the markdown file will get removed.

You should be able to check this the following way:

1. open a text file with one single editing client
2. add an attachment file
3. remove the attachment in the editor, see that it's still there in the `.attachments.<fileid>` folder
4. close the editing client
5. wait until the next Nextcloud cronjob runs
6. see that the attachment is gone from the `.attachments.<fileid>` folder

**Comment by MrRinkana** [comment]
I don't think this works as described @mejo-, for me the attachments that were removed have stayed the two days since creation. Maybe the reason is the entire folder containing both the text.md and the .attachments? I'll test it a bit more and edit if I find something more.
